api/base/serializers.py

```diff
@@ -448,7 +448,11 @@ def get_url(self, obj, view_name, request, format):
                 else:
                     url = self.reverse(view, kwargs=kwargs, request=request, format=format)
                     if self.filter:
-                        url = '{}?filter{}'.format(url, self.format_filter(obj))
+                        formatted_filter = self.format_filter(obj)
+                        if formatted_filter:
```

Comment:
`self.filter` is `True` when the `filter` kwarg is passed into a `RelationshipField`

In this case, I'm passing in `filter={'target': 'get_file_guid'} to the files serializer's comments relationship field. Formatted filter gets the values and formats them into query params. If a file does not have a guid, `formatted_filter` is `None`  so the comments url should also be `None.`